ingestion-beam/bin/build-flex-templates

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# Build dataflow flex template with output to
+# gs://$TEMPLATE_BUCKET/$JOB_TYPE/flex-templates/$BUILD_ID.json
+set -e
+set -o pipefail
+
+PROJECT="${PROJECT:?}"
+TEMPLATE_BUCKET="${TEMPLATE_BUCKET:?}"
+BUILD_ID="${BUILD_ID:?}"
+
+for JOB_CLASS in Sink Decoder; do
+  JOB_TYPE="$(echo $JOB_CLASS | tr [:upper:] [:lower:])"
+
+  export IMAGE=gcr.io/$PROJECT/ingestion-beam/$JOB_TYPE:$BUILD_ID
+  docker-compose build --build-arg FLEX_TEMPLATE_JAVA_MAIN_CLASS=com.mozilla.telemetry.${JOB_CLASS}
+  docker-compose push
+
+  METADATA_FILE=metadata.$JOB_TYPE.json
+  docker-compose run --rm flex-template > $METADATA_FILE
+
+  gcloud dataflow flex-template build \
+    gs://$TEMPLATE_BUCKET/$JOB_TYPE/flex-templates/$BUILD_ID.json \
+    --image $IMAGE \
+    --sdk-language JAVA \
+    --metadata-file $METADATA_FILE
```

Comment:
```suggestion
    --flex-template-base-image JAVA8 \
    --metadata-file $METADATA_FILE
```

It would be good to make this explicit.

Reply:
`gcloud dataflow flex-template build` requires either `--image=IMAGE` or `--env=[ENV,..] --flex-template-base-image=FLEX_TEMPLATE_BASE_IMAGE --image-gcr-path=IMAGE_GCR_PATH --jar=[JAR,...]`, not both. The image is built in a previous command, so this provides `--image`.

Reply:
the image is pre-built because the auto-built image copies dependency jars one at a time, which in turn exceeds the maximum number of docker image layers.